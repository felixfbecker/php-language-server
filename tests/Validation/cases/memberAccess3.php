<?php

namespace MyNamespace;

class A {
    public static function getInitializer(ClassLoader $loader)
    {
        return \Closure::bind(function () use ($loader) {
            $loader->prefixesPsr0 = ComposerStaticInitIncludePath::$prefixesPsr0;

        }, null, ClassLoader::class);
    }
}
