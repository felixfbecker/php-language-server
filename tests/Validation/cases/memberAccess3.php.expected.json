{
    "references": {
        "MyNamespace\\ClassLoader": [
            "./memberAccess3.php"
        ],
        "Closure::bind()": [
            "./memberAccess3.php"
        ],
        "Closure": [
            "./memberAccess3.php"
        ],
        "MyNamespace\\ClassLoader->prefixesPsr0": [
            "./memberAccess3.php"
        ],
        "MyNamespace\\ComposerStaticInitIncludePath": [
            "./memberAccess3.php"
        ],
        "MyNamespace\\ComposerStaticInitIncludePath::$prefixesPsr0": [
            "./memberAccess3.php"
        ],
        "MyNamespace\\ClassLoader::class": [
            "./memberAccess3.php"
        ]
    },
    "definitions": {
        "MyNamespace": {
            "fqn": "MyNamespace",
            "extends": [],
            "isMember": false,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "MyNamespace",
                "kind": 3,
                "location": {
                    "uri": "./memberAccess3.php"
                },
                "containerName": ""
            },
            "type": null,
            "declarationLine": "namespace MyNamespace;",
            "documentation": null,
            "signatureInformation": null
        },
        "MyNamespace\\A": {
            "fqn": "MyNamespace\\A",
            "extends": [],
            "isMember": false,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": true,
            "symbolInformation": {
                "name": "A",
                "kind": 5,
                "location": {
                    "uri": "./memberAccess3.php"
                },
                "containerName": "MyNamespace"
            },
            "type": null,
            "declarationLine": "class A {",
            "documentation": null,
            "signatureInformation": null
        },
        "MyNamespace\\A::getInitializer()": {
            "fqn": "MyNamespace\\A::getInitializer()",
            "extends": [],
            "isMember": true,
            "roamed": false,
            "isStatic": true,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "getInitializer",
                "kind": 6,
                "location": {
                    "uri": "./memberAccess3.php"
                },
                "containerName": "MyNamespace\\A"
            },
            "type__tostring": "mixed",
            "type": {},
            "declarationLine": "public static function getInitializer(ClassLoader $loader)",
            "documentation": null,
            "signatureInformation": {
                "label": "(\\MyNamespace\\ClassLoader $loader)",
                "documentation": null,
                "parameters": [
                    {
                        "label": "\\MyNamespace\\ClassLoader $loader",
                        "documentation": null
                    }
                ]
            }
        }
    }
}